bazel_tools/client_server/client_server_test.bzl

```diff
@@ -12,25 +12,35 @@ def _client_server_test_impl(ctx):
         output = wrapper,
         content = """#!/usr/bin/env bash
 set -eou pipefail
+canonicalize_rlocation() {{
+  # Note (MK): This is a fun one: Let’s say $TEST_WORKSPACE is "compatibility"
+  # and the argument points to a target from an external workspace, e.g.,
+  # @daml-sdk-0.0.0//:daml. Then the short path will point to
+  # ../daml-sdk-0.0.0/daml. Putting things together we end up with
+  # compatibility/../daml-sdk-0.0.0/daml. On Linux and MacOS this works
+  # just fine. However, on windows we need to normalize the path
+  # or rlocation will fail to find the path in the manifest file.
+  rlocation $(realpath -L -s -m --relative-to=$PWD $TEST_WORKSPACE/$1)
```

Comment:
Yes, this is a very unfortunate aspect of the runfiles mechanism. Another solution is to pass in the rootpath instead of the short path. Sadly that requires extra hoops to jump through. In this case it looks like all those are `Target`s and have `Label`s. So, you could call [`expand_location`](https://docs.bazel.build/versions/master/skylark/lib/ctx.html#expand_location) to obtain the `$(rootpath)`.

Reply:
I’ll leave it for now since this seems to work and I’m scared to touch it now :fearful: